src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -2713,6 +2713,25 @@ float Unit::GetUnitCriticalChance(WeaponAttackType attackType, Unit const* victi
         if (!(ToCreature()->GetCreatureTemplate()->flags_extra & CREATURE_FLAG_EXTRA_NO_CRIT))
         {
             chance = 5.0f;
+
+            if (IsPet() || IsHunterPet() || IsGuardian())
+            {
+                switch (attackType)
+                {
+                case BASE_ATTACK:
```

Comment:
Formatting

Reply:
Not necessarily. `case` is a label, not a function in itself.
Can be indented if you choose to, but not required.